src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +13,400 @@
+        generalSettings.setGiverParticipantOptions(getParticipantOptions(question, true));
+        generalSettings.setRecipientParticipantOptions(getParticipantOptions(question, false));
+        
+        
+        // maps for setting visibility
+        Map<String, Boolean> isGiverNameVisible = new HashMap<String, Boolean>();
+        for (FeedbackParticipantType giverType : question.showGiverNameTo) {
+            isGiverNameVisible.put(giverType.name(), true);
+        }
+        generalSettings.setGiverNameVisibleFor(isGiverNameVisible);
+        
+        Map<String, Boolean> isRecipientNameVisible = new HashMap<String, Boolean>();
+        for (FeedbackParticipantType recipientType : question.showRecipientNameTo) {
+            isRecipientNameVisible.put(recipientType.name(), true);
+        }
+        generalSettings.setRecipientNameVisibleFor(isRecipientNameVisible);
+        
+        Map<String, Boolean> isResponsesVisible = new HashMap<String, Boolean>();
+        for (FeedbackParticipantType participantType : question.showResponsesTo) {
+            isResponsesVisible.put(participantType.name(), true);
+        }
+        generalSettings.setResponseVisibleFor(isResponsesVisible);
+        
+        generalSettings.setNumberOfEntitiesToGiveFeedbackToChecked(question.numberOfEntitiesToGiveFeedbackTo == Const.MAX_POSSIBLE_RECIPIENTS);
+        generalSettings.setNumOfEntitiesToGiveFeedbackToValue(question.numberOfEntitiesToGiveFeedbackTo == Const.MAX_POSSIBLE_RECIPIENTS ?
+                                                                 1 :
+                                                                 question.numberOfEntitiesToGiveFeedbackTo);
+        qnForm.setQuestionHasResponses(questionHasResponses.get(question.getId()));
+        
+        List<String> visibilityMessages = question.getVisibilityMessage();
+        generalSettings.setVisibilityMessages(visibilityMessages);
+        
+        qnForm.setQuestionSpecificEditFormHtml(question.getQuestionDetails().getQuestionSpecificEditFormHtml(question.questionNumber));
+        qnForm.setEditable(false);
+        
+        qnForms.add(qnForm);
+    }
+
+
+    private void buildNewQuestionForm(FeedbackSessionAttributes feedbackSession,
+                                      List<FeedbackQuestionAttributes> questions,
+                                      InstructorAttributes instructor) {
+        newQnForm = new FeedbackQuestionEditForm();
+        newQnForm.setDoneEditingLink(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                                        .withUserId(account.googleId)
+                                        .withCourseId(feedbackSession.courseId)
+                                        .withSessionName(feedbackSession.feedbackSessionName));
+        
+        
+        newQnForm.setAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_ADD));
+        newQnForm.setCourseId(instructor.courseId);
```

Comment:
Is there any difference between `feedbackSession.courseId` and `instructor.courseId`?
This is the only place where `instructor` is being used in this method.